### PR TITLE
feat: Phone number based automation conditions

### PR DESCRIPTION
**Pull request by tejaswinichile**
Fixes: https://linear.app/chatwoot/issue/CW-1427/run-automation-on-contactphone-number

https://www.loom.com/share/2ea7b683efa3455283e21ca7f4e1b321

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #  description           :text
 #  meta                  :jsonb
 #  position              :integer
+#  sequence              :integer
+#  sequence              :integer
```
Comment (pranavrajs) [inline app/models/article.rb:11]: ?

**Comment by pranavrajs** [comment]
@tejaswinichile StartsWith filter is missing.

**Comment by tejaswinichile** [comment]
Example of phone number starts with filter:

https://www.loom.com/share/10c82f8eaa2d40db8c174eff26fb8d61

File: spec/services/automation_rules/conditions_filter_service_spec.rb [hunk 2]
```diff
@@ -6,21 +6,45 @@
...
+          expect(described_class.new(rule, conversation, { changed_attributes: { status: %w[open resolved] } }).perform).to be(false)
+        end
       end
     end
 
-    context 'when conditions in rule does not match with object' do
-      it 'will return false' do
-        conversation.update(status: 'resolved')
-        expect(described_class.new(rule, conversation, { changed_attributes: { status: %w[open resolved] } }).perform).to be(false)
+    context 'when conditions based on phone_number' do
```
Comment (sojan-official) [inline spec/services/automation_rules/conditions_filter_service_spec.rb:29]:
```suggestion
    context 'when conditions based on filter_operator start_with' do
```

File: spec/services/automation_rules/conditions_filter_service_spec.rb [hunk 3]
```diff
@@ -6,21 +6,45 @@
   let(:rule) { create(:automation_rule, account: account) }
 
   describe '#perform' do
-    before do
-      rule.conditions = [{ 'values': ['open'], 'attribute_key': 'status', 'query_operator': nil, 'filter_operator': 'equal_to' }]
-      rule.save
-    end
+    context 'when conditions based on status' do
```
Comment (sojan-official) [inline spec/services/automation_rules/conditions_filter_service_spec.rb:9]:
```suggestion
    context 'when conditions based on filter_operator equal_to' do
```

**Review by sojan-official — COMMENTED** [review]
LGTM overall.  

note: for the conditions_filter_service  tests, lets focus our tests on `filter_operators` and `query_operators` . Since that's where we would want to ensure stability. There wouldn't be significant behaviour change based on attributes, as only the value matters.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**